internal/backend/runtime/omni/state_validation.go

```diff
@@ -1119,3 +1119,97 @@ func nodeForceDestroyRequestValidationOptions(st state.State) []validated.StateO
 		})),
 	}
 }
+
+const (
+	// tsgen:MaxJoinTokenNameLength
+	maxJoinTokenNameLength = 16
+	// tsgen:MinJoinTokenLength
+	minJoinTokenLength = 8
+)
+
+func joinTokenValidationOptions(st state.State) []validated.StateOption {
+	validateJoinTokenName := func(res *siderolink.JoinToken) error {
+		if res.TypedSpec().Value.Name == "" {
+			return errors.New("the join token name cannot be empty")
+		}
+
+		if len(res.TypedSpec().Value.Name) > maxJoinTokenNameLength {
+			return fmt.Errorf("join token name cannot be longer than %d symbols", maxJoinTokenNameLength)
+		}
+
+		return nil
+	}
+
+	checkDefault := func(ctx context.Context, id string) (bool, error) {
+		defaultJoinToken, err := safe.ReaderGetByID[*siderolink.DefaultJoinToken](ctx, st, siderolink.DefaultJoinTokenID)
+		if err != nil && !state.IsNotFoundError(err) {
+			return false, err
+		}
+
+		if defaultJoinToken == nil {
+			return false, nil
+		}
+
+		return defaultJoinToken.TypedSpec().Value.TokenId == id, nil
+	}
+
+	return []validated.StateOption{
+		validated.WithUpdateValidations(validated.NewUpdateValidationForType(func(_ context.Context, old, res *siderolink.JoinToken, _ ...state.UpdateOption) error {
+			if old.TypedSpec().Value.Name == res.TypedSpec().Value.Name {
+				return nil
+			}
+
+			return validateJoinTokenName(res)
+		})),
+		validated.WithCreateValidations(validated.NewCreateValidationForType(func(_ context.Context, res *siderolink.JoinToken, _ ...state.CreateOption) error {
```

Comment:
no need for the create validation for all, as the mgmt api will bypass this validation. we need to do the length validation in mgmt api.